src/equations/polytropic_euler_2d.jl

```diff
@@ -0,0 +1,204 @@
+                                       equations::PolytropicEulerEquations2D)
+        # Unpack left and right state
+        rho_ll, v1_ll, v2_ll = cons2prim(u_ll, equations)
+        rho_rr, v1_rr, v2_rr = cons2prim(u_rr, equations)
+        p_ll = equations.kappa * rho_ll^equations.gamma
+        p_rr = equations.kappa * rho_rr^equations.gamma
+        v_dot_n_ll = v1_ll * normal_direction[1] + v2_ll * normal_direction[2]
+        v_dot_n_rr = v1_rr * normal_direction[1] + v2_rr * normal_direction[2]
+
+        # Compute the necessary mean values
+        if equations.gamma == 1.0 # isothermal gas
+            rho_mean = ln_mean(rho_ll, rho_rr)
+        else # equations.gamma > 1 # polytropic gas
+            rho_mean = stolarsky_mean(rho_ll, rho_rr, equations.gamma)
+        end
+        v1_avg = 0.5 * (v1_ll + v1_rr)
+        v2_avg = 0.5 * (v2_ll + v2_rr)
+        p_avg = 0.5 * (p_ll + p_rr)
+
+        # Calculate fluxes depending on normal_direction
+        f1 = rho_mean * 0.5 * (v_dot_n_ll + v_dot_n_rr)
+        f2 = f1 * v1_avg + p_avg * normal_direction[1]
+        f3 = f1 * v2_avg + p_avg * normal_direction[2]
+
+        return SVector(f1, f2, f3)
+    end
+
+    @inline function min_max_speed_naive(u_ll, u_rr, normal_direction::AbstractVector,
+                                         equations::PolytropicEulerEquations2D)
+        rho_ll, v1_ll, v2_ll = cons2prim(u_ll, equations)
+        rho_rr, v1_rr, v2_rr = cons2prim(u_rr, equations)
+        p_ll = equations.kappa * rho_ll^equations.gamma
+        p_rr = equations.kappa * rho_rr^equations.gamma
+
+        v_normal_ll = v1_ll * normal_direction[1] + v2_ll * normal_direction[2]
+        v_normal_rr = v1_rr * normal_direction[1] + v2_rr * normal_direction[2]
+
+        norm_ = norm(normal_direction)
+        # The v_normals are already scaled by the norm
+        lambda_min = v_normal_ll - sqrt(equations.gamma * p_ll / rho_ll) * norm_
+        lambda_max = v_normal_rr + sqrt(equations.gamma * p_rr / rho_rr) * norm_
+
+        return lambda_min, lambda_max
+    end
+
+    @inline function max_abs_speeds(u, equations::PolytropicEulerEquations2D)
+        rho, v1, v2 = cons2prim(u, equations)
+        c = sqrt(equations.gamma * equations.kappa * rho^(equations.gamma - 1))
+
+        return abs(v1) + c, abs(v2) + c
+    end
+
+    # Convert conservative variables to primitive
+    @inline function cons2prim(u, equations::PolytropicEulerEquations2D)
+        rho, rho_v1, rho_v2 = u
+
+        v1 = rho_v1 / rho
+        v2 = rho_v2 / rho
+
+        return SVector(rho, v1, v2)
+    end
+
+    # Convert conservative variables to entropy
+    @inline function cons2entropy(u, equations::PolytropicEulerEquations2D)
+        rho, rho_v1, rho_v2 = u
+
+        v1 = rho_v1 / rho
+        v2 = rho_v2 / rho
+        v_square = v1^2 + v2^2
+        p = pressure(u, equations)
+        # Form of the internal energy depends on gas type
+        if equations.gamma == 1.0 # isothermal gas
```

Comment:
It would be good to add a test that exercises this code path. Right now, you only have elixirs with `gamma = 1.001` or `gamma = 2.0` if I am not mistaken. Maybe you can change one of them to `gamma = 1.0` or, alternatively, add a new test that runs an existing elixir but with a different `gamma` for a few timesteps.

Reply:
I agree. The whole purpose of having a specialized path for the isothermal case versus the polytropic case is that we do not need to use the "trick" of setting the gas gamma near one, e.g., `gamma = 1.001` to model an isothermal gas. We can use the appropriate fluxes for the special isothermal case without the worry of dividing by `gamma - 1` because it never appears is the corresponding code.

Reply:
Changed the isothermal tests to gamma =1.0.